reana_client/cli/retention_rules.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2022 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+"""REANA client retention-rules related commands."""
+
+import logging
+import sys
+from typing import Tuple
+
+import click
+
+from reana_client.api.client import get_workflow_retention_rules
+from reana_client.cli.utils import (
+    add_access_token_options,
+    add_workflow_option,
+    check_connection,
+    display_formatted_output,
+)
+from reana_client.printer import display_message
+
+
+@click.group(help="Retention rules commands")
+def retention_rules_group():
+    """Retention rules commands."""
+    pass
+
+
+@retention_rules_group.command()
+@add_workflow_option
+@add_access_token_options
+@check_connection
+@click.option(
+    "--format",
+    "_format",
+    multiple=True,
+    help="Format output according to column titles or column values. "
+    "Use `<columm_name>=<column_value>` format. "
+    "E.g. display pattern and status of active retention rules "
+    "`--format workspace_files,status=active`.",
+)
+@click.option(
+    "--json",
+    "output_format",
+    flag_value="json",
+    default=None,
+    help="Get output in JSON format.",
+)
+def retention_rules_show(
+    access_token: str, workflow: str, _format: Tuple[str], output_format: str
+) -> None:  # noqa: D301
+    """Show the retention rules of a workflow.
```

Comment:
I used `Show ...` to keep the same verb as in `retention-rules-show`. 

Taking into consideration the other commands, there is both a command with `...-show` (`quota-show`) and a command ending in `...-list` (`secrets-list`). Should we consider using `retention-rules-list`? /cc @tiborsimko

Reply:
Both could make sense: 

- OT1H, I think "-show" is probably clearer here, since we operate on the rules list as a whole for now, kind of like quota.  (And the term show nicely differentiates from the workflow list, so there is no possible confusion about the word "list" here.)

- OTOH, "-list" would make sense if we allow operating on a single rule in the future, as in "keep those 7 rules as they are and extend these 2 rules for me please". 

If we are going to provide the "-extend" functionality as a whole, then "-show" might make sense. If we are going to provide it for each individual rule separately, then "-list' might make sense. So, we could use it already now, depending on future possible evolution.

(The only concern is not to create confusion about the meaning of "-list" and what is the relation of the  "workflow list" to the "rule list", so to speak.)

Perhaps "-list" is more flexible WRT future?

P.S. I would also call it "Show retention rules for a workflow", not "...of a workflow", since they are kind of ancillary to the workflow research nature, and could be shortened/extended independently of the workflow run status, etc.